### ISSUE TITLE
[Wayland] Dragging items in toolbox causes toolbox item to float off-position

### ISSUE THREAD

**Issue by vchernin** [tenacityteam/tenacity-legacy#563]
### Guidelines

- [X] I have read the guidelines.

### Version/Commit hash

https://github.com/flathub/flathub/pull/2444#issuecomment-907707785 (no commit hash with Flatpak apparently)

### Describe the bug.

1. Ensure Tenacity is being run in Native Wayland mode.
2. Drag an item in the toolbar around (e.g. the play/pause buttons etc.), and try to move it.
3. You will notice as soon as you detach it the toolbar items randomly decides to get stuck somewhere.
4. Once you've placed the item down again it will return to a reasonable position.

https://user-images.githubusercontent.com/38842733/131316270-0a9d7bce-59db-42e6-8da1-90ee1ea90e11.mp4

### Expected behavior

1. Dragging and dropping works as expected, and does not cause an item in the toolbar to float where it shouldn't be.

### OS

Linux

### This issue is not a duplicate

- [X] I have confirmed this issue isn't a duplicate.

**Comment by nbsp** [comment]
Also broken on Sway, using current HEAD (1b2c9038a4bec69ebaf07846fa007583a34c23d0)

**Comment by leio** [comment]
This isn't really something we can do with wayland at all in this undockable form nicely. What we can do is keep it within the toplevel window itself, and never undock like that (which is rather questionable in the first place and often done accidentally imo). See LibreOffice for an example, that's not very good yet either, but feels a lot better at least. Something like https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/46 might help make that better or support what it tries to do on Xorg (didn't read into it very deeply yet)

**Comment by akleja** [comment]
I agree with keeping it in the toplevel window. Creating new windows just to display an indicator for the docking positions seems really unnecessary and behaves badly in other tiling window managers as well (at least for me in xmonad and awesome wm)

**Comment by lanodan** [comment]
> I agree with keeping it in the toplevel window. Creating new windows just to display an indicator for the docking positions seems really unnecessary and behaves badly in other tiling window managers as well (at least for me in xmonad and awesome wm)

Except wayland gives the ability to explicitly define windows as popups so misbehavior that are quite typical in X11 shouldn't exists if the window is created correctly. And I can confirm that even if not positioned with the cursor, it is a popup (tested tenacity in sway and cage, native wayland).
